src/dashboard/apps/consent/management/commands/renewconsents.py

```diff
@@ -12,18 +12,27 @@ class Command(DashboardBaseCommand):
     help = __doc__
 
     def handle(self, *args, **options):
-        """Renew expiring consents that are nearing expiration.
+        """Renew expiring consents that are nearing expiration and generate new ones.
 
-        This method attempts to duplicate expiring
-        consents and logs an error message if the operation fails.
+        This method attempts to
+        - duplicate expiring consents and logs an error message if the operation fails.
+        - generate new consents for delivery points that do not have active consents.
 
         Raises:
             Exception: Captures any exception occurring during the consent
             duplication process and logs the error, including the relevant SIRET and
             error details.
         """
         try:
-            renew_expiring_consents()
+            expiring_consents = renew_expiring_consents()
+            self._log_notice(f"Duplicate {len(expiring_consents)} consent(s).")
```

Comment:
Should be outside from the try/except block:

```suggestion
            self._log_notice(f"Duplicate {len(expiring_consents)} consent(s).")
```